p2p/proto/block.proto

```diff
@@ -0,0 +1,66 @@
+syntax = "proto3";
+import "p2p/proto/common.proto";
+import "google/protobuf/timestamp.proto";
+
+// for now, we assume a small consensus, so this fits in 1M. Else, these will be repeated
+message Signatures {
+    BlockID id = 1;
+    repeated Signature signatures = 2; // can be more explicit here about the signature structure as this is not part of account abstraction
+}
+
+// Note: commitments may change to be for the previous blocks like comet/tendermint
+// hash of block header sent to L1
+message BlockHeader {
+    BlockID parent_block = 1;
+
+    google.protobuf.Timestamp time = 2; // TODO: see if this needs to be Felt252 or can be converted
+
+    Address sequencer_address = 3;
+
+    Merkle state_diffs = 4; //  By order of (contract, key), taking last in case of duplicates.
+                            // This means the proposer needs to sort after finishing the block (TBD: patricia? )
+
+    // State is optional and appears every X blocks for the last block. This is to support
+    // snapshot sync and also so that light nodes can sync on state without state diffs.
+    Merkle state = 5; // hash of contract and class patricia tries. Same as in L1. Later more trees will be included
+
+    Hash proof_fact = 6; // for Kth block behind. A hash of the output of the proof
+
+    // The following merkles can be built on the fly while sequencing/validating txs.
+    Merkle transactions = 7; // By order of execution
+
+    Merkle events = 8; // By order of issuance.
+
+    Merkle receipts = 9; // By order of issuance.
+
+    uint32 protocol_version = 10;
+
+    ChainID chain_id = 11;
+}
+
+message BlockProof {
+    bytes proof = 1; // proof size is currently 142K
+}
+
+// sent to all peers (except the ones this was received from, if any).
+// for a fraction of peers, also send the GetBlocks response (as if they asked for it for this block)
+message NewBlock {
+    BlockID id = 1;
+}
+
+// result is (BlockHeader, StateDiff*)*
+message GetBlocks {
+    enum Direction {
+        Forward = 0;
+        Backward = 1;
+    }
+    BlockID start = 1; // exclude start from the result
+    Direction direction = 2;
+    uint64 limit = 3;
+    uint64 skip = 4;  // when starting, can send to first node skip=0, limit=100, second skip=100, limit=100, etc.
+    uint64 step = 5; // to allow interleaving from several nodes
```

Comment:
```suggestion
    optional uint64 step = 5; // to allow interleaving from several nodes
```

The real default `step` is `1`, but if the field is missing, it will be assigned a default value of `uint64` which is `0` and does not make sense.

Reply:
so 'step' will be 0 if not provided in the iteration. but then limit can also be 0 and direction null. Seems awkward to me to make 'step' optional while the rest are not. Let the client always be explicit about step ust return an error (need to define a message for that?) if step = 0.